test/yargs-parser.cjs

```diff
@@ -3587,6 +3589,121 @@ describe('yargs-parser', function () {
+      const args = parser("--foo $'text with \\n newline'")
+      args.foo.should.equal('text with \n newline')
+
+      // Double quotes shouldn't work
+      const args2 = parser('--foo $"text without \\n newline"')
+      args2.foo.should.equal('$"text without \\n newline"')
+
+      const characters = '\\\\' + '\\a' + '\\b' + '\\e' + '\\E' + '\\f' + '\\n' + '\\r' + '\\t' + '\\v' + "\\'" + '\\"' + '\\?'
+      const args3 = parser("--foo $'" + characters + "'")
+      args3.foo.should.equal('\\\a\b\u001b\u001b\f\n\r\t\v\'"?') // eslint-disable-line
+
+      const args4 = parser("--foo $'text \\xFFFF with \\xFF hex'")
+      args4.foo.should.equal('text \u00FFFF with \u00FF hex')
+      const args5 = parser("--foo $'text \\uFFFFFF\\uFFFF with \\uFF hex'")
+      args5.foo.should.equal('text \uFFFFFF\uFFFF with \u00FF hex')
+      const args6 = parser("--foo $'text \\U10FFFF\\UFFFF with \\U00FF hex'")
+      const longCodePoint = String.fromCodePoint(0x10FFFF)
+      args6.foo.should.equal(`text ${longCodePoint}\uFFFF with \u00FF hex`)
+
+      const args7 = parser("--foo $'text \\cAB \\cz with \\c12 control \\c011 chars'")
+      args7.foo.should.equal('text \u0001B \u001A with \u00112 control \u001011 chars')
+
+      const args8 = parser("--foo $'text \\0 \\001 with \\12 \\123 \\129 octal'")
+      args8.foo.should.equal('text \u0000 \u0001 with \u000A \u0053 \u000A9 octal')
+    })
+
+    function hexEncode (s) {
+      let hex
+      let result = ''
+      for (let i = 0; i < s.length; i++) {
+        hex = s.charCodeAt(i).toString(16)
+        result += ('000' + hex).slice(-4)
+      }
+      return result
+    }
+
+    async function check (ansiString, encoding = 'utf8') {
+      const result = parser(['--foo', ansiString]).foo + '' // convert integers back to str
+      const resultBuffer = Buffer.from(result, encoding)
+
+      const { stdout } = await exec(
+        'printf "%s" ' + ansiString,
+        { shell: '/bin/bash', encoding: 'buffer' }
+      )
+      if (!resultBuffer.equals(stdout)) {
+        console.log(JSON.stringify(ansiString), hexEncode(result), result.length, resultBuffer, resultBuffer.length, stdout, stdout.length)
+      }
+      resultBuffer.should.deep.equal(stdout)
+    }
+
+    async function checkAll (prefix, value, maxLength, encoding = 'utf8') {
+      await check("$'\\" + prefix + value + "'", encoding)
+      while (value.length < maxLength) {
+        value = '0' + value
+        await check("$'\\" + prefix + value + "'", encoding)
+        if (value.toLowerCase() !== value) {
+          await check("$'\\" + prefix + value.toLowerCase() + "'", encoding)
+        }
+      }
+    }
+
+    // This passes
+    xit('parses all octal codes in ANSI-C quoted strings like bash', async () => {
+      for (let i = 0; i <= 0o777; i++) {
+        if (i % 256 === 0) { // null character
+          continue
+        }
+        await checkAll('', i.toString(8), 3, 'ascii')
+      }
+    }).timeout(5000)
```

Comment:
I'm surprised by the need to extend the timeout, the parsing of strings should be quite fast.

Reply:
This test runs a bash command through `exec` 511 times. On my desktop it completes in slightly under the default 2 second timeout, but I added some headroom.

Reply:
I removed these tests, they were more to help me understand how bash does the parsing while I was working on this.